libs/wait-for-ecs-deploy/src/dynamic-resource-provider.spec.ts

```diff
@@ -0,0 +1,214 @@
+            'ecs-describe-services-all-services-deployed.json',
+        )
+        const { json, clusters, serviceCount } = getSampleResponseFromFile(file)
+        expect.assertions(serviceCount)
+        ECS.mockImplementation(mockedEcs(json))
+
+        for (const cluster of clusters) {
+            for (const service of cluster.services) {
+                const result = waitForService({
+                    clusterName: cluster.clusterName,
+                    serviceName: service.serviceName,
+                    desiredTaskDef: service.taskDefinition,
+                })
+                await expect(result).resolves.toHaveProperty(
+                    'status',
+                    'COMPLETED',
+                )
+            }
+        }
+    })
+
+    it('should return FAILED when one or more services has failed and a rollback is in progress', async () => {
+        const file = path.join(
+            __dirname,
+            '__mocks__',
+            'ecs-describe-services-failed-rollback-in-progress.json',
+        )
+        const { json, clusters, serviceCount } = getSampleResponseFromFile(file)
+        expect.assertions(serviceCount)
+        ECS.mockImplementation(mockedEcs(json))
+
+        for (const cluster of clusters) {
+            for (const service of cluster.services) {
+                const result = waitForService({
+                    clusterName: cluster.clusterName,
+                    serviceName: service.serviceName,
+                    desiredTaskDef: service.taskDefinition,
+                })
+                await expect(result).resolves.toHaveProperty('status', 'FAILED')
+            }
+        }
+    })
+
+    it('should return FAILED when a rollback has completed', async () => {
+        const file = path.join(
+            __dirname,
+            '__mocks__',
+            'ecs-describe-services-failed-rolled-back.json',
+        )
+        const { json, clusters, serviceCount } = getSampleResponseFromFile(file)
+        expect.assertions(serviceCount)
+        ECS.mockImplementation(mockedEcs(json))
+
+        for (const cluster of clusters) {
+            for (const service of cluster.services) {
+                const result = waitForService({
+                    clusterName: cluster.clusterName,
+                    serviceName: service.serviceName,
+                    desiredTaskDef: `${service.taskDefinition}2`,
+                })
+                await expect(result).resolves.toHaveProperty('status', 'FAILED')
+            }
+        }
+    })
+
+    it('should time out if it takes too long', async () => {
+        const file = path.join(
+            __dirname,
+            '__mocks__',
+            'ecs-describe-services-failed-rollback-in-progress.json',
+        )
+        const { clusters, serviceCount } = getSampleResponseFromFile(file)
+        expect.assertions(serviceCount)
+        ECS.mockImplementation(mockedEcsDeploymentTimeout())
+        jest.useFakeTimers()
```

Comment:
This isn't a good idea, it affects promise resolution and a whole bunch of other things. I'd imagine it would affect the behavior of pulumi a fair bit

Reply:
Interesting, I hadn't thought of that. The only timer than needs to be mocked is the timeout timer, but it could probably just be a very short timeout instead.

Reply:
Ok, I've removed `jest.useFakeTimers()`, using a 10ms timeout instead.